dotnet/src/SemanticKernel.Core/Planning/Plan.cs

```diff
@@ -79,7 +80,7 @@ public sealed class Plan : ISKFunction
 
     /// <inheritdoc/>
     [JsonPropertyName("model_settings")]
-    public List<AIRequestSettings>? ModelSettings { get; private set; }
+    public ReadOnlyCollection<AIRequestSettings> ModelSettings => this.Function is not null ? this.Function.ModelSettings : Array.Empty<AIRequestSettings>().ToList<AIRequestSettings>().AsReadOnly();
```

Comment:
Array is ReadONly, you can use ?? to simplify the line. 😊
```suggestion
    public ReadOnlyCollection<AIRequestSettings> ModelSettings => this.Function?.ModelSettings ?? Array.Empty<AIRequestSettings>();
```

Reply:
Done